src/core/render-biblio.js

```diff
@@ -107,27 +107,37 @@ export function run(conf) {
  * and warns about circular references
  * @param {String} ref
  */
-function getRefContent(conf) {
-  return function(ref) {
-    let refcontent = conf.biblio[ref];
-    let key = ref;
-    const circular = new Set([key]);
-    while (refcontent && refcontent.aliasOf) {
-      if (circular.has(refcontent.aliasOf)) {
-        refcontent = null;
-        const msg = `Circular reference in biblio DB between [\`${ref}\`] and [\`${key}\`].`;
-        pub("error", msg);
-      } else {
-        key = refcontent.aliasOf;
-        refcontent = conf.biblio[key];
-        circular.add(key);
-      }
-    }
-    if (refcontent && !refcontent.id) {
-      refcontent.id = ref.toLowerCase();
+function toRefContent(ref) {
+  let refcontent = biblio[ref];
+  let key = ref;
+  const circular = new Set([key]);
+  while (refcontent && refcontent.aliasOf) {
+    if (circular.has(refcontent.aliasOf)) {
+      refcontent = null;
+      const msg = `Circular reference in biblio DB between [\`${ref}\`] and [\`${key}\`].`;
+      pub("error", msg);
+    } else {
+      key = refcontent.aliasOf;
+      refcontent = biblio[key];
+      circular.add(key);
     }
-    return { ref, refcontent };
-  };
+  }
+  if (refcontent && !refcontent.id) {
+    refcontent.id = ref.toLowerCase();
+  }
+  return { ref, refcontent };
+}
+
+/**
+ * Render an inline citation
+ *
+ * @param {String} ref the inline reference.
+ * @returns HTMLElement
+ */
+export function renderInlineCitation(ref) {
```

Comment:
Wait, but what's this?

Reply:
Sorry, should have done this as different hunk. We then use this in the next part to render inline citations.